src/utils/convert3Dto4D.m

```diff
@@ -0,0 +1,192 @@
+optSource.dataType = 0;
+
+% Get source folder content
+sourceDataStruc = dir(optSource.sourceDir);
+
+isDir = [sourceDataStruc(:).isdir];
+
+optSource.sequenceList = {sourceDataStruc(isDir).name}';
+
+%% Loop through the sequence folders
+
+tic
+
+for iSeq = 1:size(optSource.sequenceList, 1)
+
+    % Skip 'non' foldera
+    if length(char(optSource.sequenceList(iSeq))) > 2
+
+        % Check if sequence to ignore or not
+        if sum(strcmp(optSource.sequenceList(iSeq), optSource.sequenceToIgnore)) == 1
+
+            newline;
+            warning('IGNORING SEQUENCE: %s', optSource.sequenceToIgnore{strcmp(optSource.sequenceList(iSeq), optSource.sequenceToIgnore)})
+            newline;
+
+        else
+
+            fprintf('\n\nCONVERTING SEQUENCE: %s \n\n', char(optSource.sequenceList(iSeq)))
+
+            % Get sequence folder path
+            sequencePath = char(fullfile(optSource.sourceDir, optSource.sequenceList(iSeq)));
+
+            % Retrieve volume files info
+            [volumesToConvert, volumesList] = parseNiiFiles(sequencePath, optSource);
+
+            % Set output name, it takes the file name of the first volume and add subfix '_4D'
+            outputName = char(volumesToConvert(1).name);
+
+            dotPos = find(outputName == '.');
+
+            outputName = outputName(1:dotPos(1)-1);
+
+            optSource.outputName = [outputName '_4D.nii' ];
+
+            % Retrieve sidecar json files info
+            jsonList = parseJsonFiles(sequencePath);
+
+            jsonFile = jsondecode(fileread(char(jsonList(1))));
+
+            % % % % % % LIEGE  SPECIFIC % % % % % % %
+            optSource.RT = jsonFile.acqpar.RepetitionTime/1000;
+            % % % % % % % % % % % % % % % % % % % % %
+
+            % Set and run spm batch
+            matlabbatch = setBatch3Dto4D(optSource, volumesList);
+
+            spm_jobman('run', matlabbatch);
+
+            % Zip and delete the and the new 4D file
+            fprintf(1, 'ZIP AND DELETE THE NEW 4D BRAIN \n\n');
+
+            gzip([sequencePath filesep outputName])
+
+            delete([sequencePath filesep outputName])
+
+            % Save one sidecar json file, it takes the first volume one and add subfix '_4D'
+            dotPos = find(char(jsonList(1)) == '.', 1, 'last');
+
+            jsonCopy = char(jsonList(1));
+
+            jsonCopy = jsonCopy(1:dotPos(1)-1);
+
+            copyfile(char(jsonList(1)), [jsonCopy '_4D.json' ])
+
+            % Delete all the single volumes .nii and .json files
+            fprintf(1, 'EXTERMINATE SINGLE VOLUMES FILES \n\n');
+
+            for iDel = 1:length(optSourcevolumesList)
+
+                delete(char(optSourcevolumesList(iDel)))
+                delete(char(jsonList(iDel)))
+
+            end
+
+        end
+
+    end
+
+end
+
+newline;
+toc
+
+end
+
+function [volumesToConvert, optSource] = parseNiiFiles(sequencePath, optSource)
+
+volumesToConvert = dir(fullfile(sequencePath, '*.nii'));
```

Comment:
if you don't want to loop through things maybe have a look into `spm_select` to give you the fullpath list of files in a directory that match a certain pattern.

```
spm_select('FPlist', sequencePath,  '^*.nii$')
```